### PR TITLE
feat: 847 update collateral tab

### LINKED ISSUE #847: Portfolio: Update Collateral tab
### Current implementation

image

### Proposed design
https://www.figma.com/file/K49Nx7XfocShhHcwwQzQkm/Console-V2?node-id=2537%3A32178

image

image

### Details
- Show 'General' margin as top level, expand to see 'Margin' (position) level margin
- General/Available margin
  - Show total margin balance
  - Show used and available margin levels
  - Add Used -> Available visualisation
  - Add buttons to deposit/withdraw collateral
- Position margin
  - Show market name, linked to market on Trading page
  - Show allocated margin value
  - Add visualisation representing how much margin is allocated to that position
  - Add button to close position
- Add button to deposit a new asset type

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #847 & #1366

# Description ℹ️

Update the collaterals tab.

# Demo 📺

After: 
Screenshot 2022-09-26 at 16 49 55
Screenshot 2022-09-26 at 16 50 15

Screenshot 2022-09-22 at 23 36 22

Screenshot 2022-09-22 at 23 36 35
Screenshot 2022-09-22 at 23 37 20

Screenshot 2022-09-22 at 21 22 56

Before:
Screenshot 2022-09-14 at 20 06 08
Screenshot 2022-09-14 at 20 06 44

190626313-d840ee6a-67d3-4283-8245-ca4e636e5148

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 1]
```diff
@@ -67,3 +94,46 @@ export const accountsDataProvider = makeDataProvider<
...
+    const incoming = assetData
+      .filter((a) => IN_ACCOUNT_TYPES.includes(a.type))
+      .reduce((acc, a) => acc + BigInt(a.balance), BigInt(0));
+
+    const used = assetData
+      .filter((a) => OUT_ACCOUNT_TYPES.includes(a.type))
+      .reduce((acc, a) => acc + BigInt(a.balance), BigInt(0));
+
+    const depositRow: AccountFields = {
+      ...assetData[0],
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts:128]: we also need  a.asset and a.market, AccountFields extends AccountFieldsFragment

File: apps/trading/pages/liquidity/[marketId].page.tsx [hunk 2]
```diff
@@ -69,9 +69,10 @@ const LiquidityPage = ({ id }: { id?: string }) => {
       <div className="h-full grid grid-rows-[min-content_1fr]">
         <Header
           title={
-            <button onClick={() => push(`/markets/${marketId}`)}>{`${name} ${t(
-              'liquidity provision'
-            )}`}</button>
+            <button
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/trading/pages/liquidity/[marketId].page.tsx]: updated

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 3]
```diff
@@ -67,3 +94,46 @@ export const accountsDataProvider = makeDataProvider<
...
+      available: (incoming - used).toString(),
+      deposited: deposited.toString(),
+      used: used.toString(),
+    };
+
+    const accountRows = assetData
+      .filter((a) => ![AccountType.ACCOUNT_TYPE_GENERAL].includes(a.type))
+      .map((a) => ({
+        ...a,
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts:139]: hm, I can be more explicit but there is more than balance, `AccountFields extends AccountFieldsFragment`

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 4]
```diff
@@ -67,3 +94,46 @@ export const accountsDataProvider = makeDataProvider<
...
+
+    const depositRow: AccountFields = {
+      ...assetData[0],
+      available: (incoming - used).toString(),
+      deposited: deposited.toString(),
+      used: used.toString(),
+    };
+
+    const accountRows = assetData
+      .filter((a) => ![AccountType.ACCOUNT_TYPE_GENERAL].includes(a.type))
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts]: updated

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 5]
```diff
@@ -67,3 +94,46 @@ export const accountsDataProvider = makeDataProvider<
...
+  const collateralData = getSymbols(data).map((assetSymbol) => {
+    const assetData = data.filter((a) => a.asset.symbol === assetSymbol);
+
+    const deposited = assetData
+      .filter((a) => [AccountType.ACCOUNT_TYPE_GENERAL].includes(a.type))
+      .reduce((acc, a) => acc + BigInt(a.balance), BigInt(0));
+
+    const incoming = assetData
+      .filter((a) => IN_ACCOUNT_TYPES.includes(a.type))
+      .reduce((acc, a) => acc + BigInt(a.balance), BigInt(0));
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts]: of course, updated

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 6]
```diff
@@ -67,3 +94,46 @@ export const accountsDataProvider = makeDataProvider<
...
+  Array.from(new Set(data.map((a) => a.asset.symbol))).sort();
+
+export const getAccountData = (
+  data: AccountFieldsFragment[]
+): AccountFields[] => {
+  const collateralData = getSymbols(data).map((assetSymbol) => {
+    const assetData = data.filter((a) => a.asset.symbol === assetSymbol);
+
+    const deposited = assetData
+      .filter((a) => [AccountType.ACCOUNT_TYPE_GENERAL].includes(a.type))
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts]: I thought I should add more in that array but there's no need to, updated

File: libs/accounts/src/lib/accounts-data-provider.spec.ts [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+    asset: {
+      __typename: 'Asset',
+      id: '5cfa87844724df6069b94e4c8a6f03af21907d7bc251593d08e4251043ee9f7c',
+      symbol: 'tBTC',
+      decimals: 5,
+    },
+  },
+];
+
+const accountResult = [
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.spec.ts]: added type.

File: libs/deposits/src/lib/deposit-form.tsx [hunk 8]
```diff
@@ -151,7 +151,11 @@ export const DepositForm = ({
         <Controller
           control={control}
           name="asset"
-          rules={{ validate: { required } }}
+          rules={{
+            validate: {
+              required: (value) => !!selectedAsset || required(value),
```
Comment (MadalinaRaicu) [inline libs/deposits/src/lib/deposit-form.tsx:156]: @mattrussell36  I do need this check here. the asset can be selected and the value would still be undefined. defaultValue set up is not sufficent.

File: libs/positions/src/lib/positions-table.tsx [hunk 9]
```diff
@@ -64,33 +66,6 @@ export const MarketNameCell = ({ valueFormatted }: MarketNameCellProps) => {
   return (valueFormatted && valueFormatted[0]) || undefined;
 };
 
-export interface PriceCellProps {
```
Comment (MadalinaRaicu) [inline libs/positions/src/lib/positions-table.tsx:67]: nothing really changed here. just moved some stuff to ui-toolkit

File: apps/trading/pages/liquidity/[marketId].page.tsx [hunk 10]
```diff
@@ -69,9 +75,12 @@ const LiquidityPage = ({ id }: { id?: string }) => {
       <div className="h-full grid grid-rows-[min-content_1fr]">
         <Header
           title={
-            <button onClick={() => push(`/markets/${marketId}`)}>{`${name} ${t(
-              'liquidity provision'
-            )}`}</button>
+            <Link href={`/markets/${marketId}`} passHref={true}>
+              <UiToolkitLink>
+                {' '}
```
Comment (MadalinaRaicu) [inline apps/trading/pages/liquidity/[marketId].page.tsx:80]:
```suggestion
```

File: libs/accounts/src/lib/breakdown-table.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+describe('BreakdownTable', () => {
+  it('should render successfully', async () => {
+    await act(async () => {
+      const { baseElement } = render(<BreakdownTable data={[]} />);
+      expect(baseElement).toBeTruthy();
+    });
+  });
+
+  it('should render correct columns', async () => {
+    act(async () => {
```
Comment (mattrussell36) [inline libs/accounts/src/lib/breakdown-table.spec.tsx:43]: chore (blocking): This is incorrect usage of `act` and there are console error warnings in test output. Please fix.

Note that these tests are also evergreen as a result.

- you must await `act(async () => ..`
- assertions should be made outside of `act` so that the component has finished rendering

File: libs/accounts/src/lib/breakdown-table.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+import type { AccountType } from '@vegaprotocol/types';
+import {
+  assetDecimalsFormatter,
+  progressBarHeaderComponentParams,
+  progressBarValueFormatter,
+} from './accounts-table';
+
+interface BreakdownTableProps extends AgGridReactProps {
+  data: AccountFields[] | null;
+  style?: CSSProperties;
```
Comment (mattrussell36) [inline libs/accounts/src/lib/breakdown-table.tsx:22]: chore (non blocking): Looks like this prop isnt used

File: libs/accounts/src/lib/index.ts [hunk 13]
```diff
@@ -1,6 +1,7 @@
-export * from './__generated___/Accounts';
-export * from './accounts-container';
+export * from './__generated__';
+// export * from './__generated___';
```
Comment (mattrussell36) [inline libs/accounts/src/lib/index.ts:2]: chore (blocking): remove

File: libs/positions/src/lib/positions-table.tsx (hunk 9 above)
Comment (mattrussell36) [reply libs/positions/src/lib/positions-table.tsx:67]: Hm if there's a circular dependency here then something is off elsewhere as the other cell components are working from there fine. Lets fix later though!

File: libs/positions/src/lib/positions-table.tsx [hunk 14]
```diff
@@ -140,14 +115,33 @@ const ButtonCell = ({
   );
 };
 
-const EmptyCell = () => '';
+const progressBarValueFormatter = ({
```
Comment (mattrussell36) [inline libs/positions/src/lib/positions-table.tsx:118]: chore (non blocking): Looks a bit like this logic is duplicated in accounts-table.tsx, Seems like the Progressbar component should handle this. Can be a separate PR

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 15]
```diff
@@ -10,54 +9,73 @@ import type {
...
     deltas.forEach((delta) => {
-      const id = getId(delta);
-      const index = draft.findIndex((a) => getId(a) === id);
+      const index = draft.findIndex(
+        (account) => account.asset.id === delta.assetId
+      );
       if (index !== -1) {
-        draft[index].balance = delta.balance;
-      } else {
-        // #TODO handle new account
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-data-provider.ts:42]: chore (non blocking): Suggest leaving this else block in

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/accounts/src/lib/accounts-table.tsx [hunk 16]
```diff
@@ -1,172 +1,187 @@
...
-  }
-
-  if (valueA > valueB) {
-    return 1;
-  }
-
-  if (nodeA.data.__summaryRow) {
-    return isInverted ? -1 : 1;
-  }
+interface AccountsTableValueFormatterParams extends ValueFormatterParams {
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx]: updated

File: libs/accounts/src/lib/accounts-table.tsx [hunk 17]
```diff
@@ -1,172 +1,187 @@
...
+  value,
+  data,
+}: AccountsTableValueFormatterParams) =>
+  addDecimalsFormatNumber(value, data.asset.decimals);
 
-  return 0;
-};
+export interface AccountTableProps extends AgGridReactProps {
+  data: AccountFields[] | null;
+  onClickAsset: (asset?: string | Asset) => void;
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx]: updated

File: libs/accounts/src/lib/accounts-table.tsx [hunk 18]
```diff
@@ -1,172 +1,187 @@
...
+            sortable: true,
+          }}
+        >
+          <AgGridColumn
+            headerName={t('Asset')}
+            field="asset.symbol"
+            headerTooltip={t(
+              'Asset is the collateral that is deposited into the Vega protocol.'
+            )}
+            cellRenderer={({ value }: ValueFormatterParams) => {
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx]: updated all

File: libs/accounts/src/lib/breakdown-table.tsx (hunk 12 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/breakdown-table.tsx]: removed

File: libs/accounts/src/lib/index.ts (hunk 13 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/index.ts]: removed

File: libs/positions/src/lib/positions-table.tsx (hunk 14 above)
Comment (MadalinaRaicu) [reply libs/positions/src/lib/positions-table.tsx:118]: it is  similar with the one in the positions-table, but the range calculation is different. I moved to ui-toolkit progress bar selectors and cell renderers.

File: libs/accounts/src/lib/accounts-data-provider.ts (hunk 15 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts:42]: updated

File: libs/accounts/src/lib/__generated___/index.ts [hunk 19]
```diff
@@ -0,0 +1 @@
+export * from './Accounts';
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/__generated___/index.ts]: updated

File: libs/accounts/src/lib/breakdown-table.spec.tsx (hunk 11 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/breakdown-table.spec.tsx:43]: I'll fix those together with the withdraw-manager spec which is timing out.

File: libs/accounts/src/lib/accounts-table.spec.tsx [hunk 20]
```diff
@@ -24,47 +25,104 @@ const singleRow: AccountFieldsFragment = {
...
 };
 const singleRowData = [singleRow];
 
 describe('AccountsTable', () => {
   it('should render successfully', async () => {
-    await act(async () => {
-      const { baseElement } = render(<AccountsTable data={[]} />);
-      expect(baseElement).toBeTruthy();
-    });
+    const { baseElement } = render(
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-table.spec.tsx:36]: chore (blocking): This test is printing out console errors, when testing components that render ag grid they need to be wrapped in act

Suggest something like the following:
```
it('should render successfully', async () => {
    await act(async () => {
      render(<AccountTable rowData={[]} onClickAsset={() => null} />);
    });
    expect(document.querySelector('.ag-root-wrapper')).toBeInTheDocument();
  });
```

File: libs/accounts/src/lib/accounts-table.spec.tsx [hunk 21]
```diff
@@ -24,47 +25,104 @@ const singleRow: AccountFieldsFragment = {
...
-          )
-        ).toEqual(['Asset', 'Type', 'Market', 'Balance']);
-      });
+    let headers: HTMLElement[];
+    await act(async () => {
+      render(
+        <AccountTable rowData={singleRowData} onClickAsset={() => null} />
+      );
+    });
+    await waitFor(async () => {
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-table.spec.tsx:49]: chore (blocking): By wrapping rendering of the grid in act you are allowing the component to finish rendering. This makes `await waitFor` redundant here

The following should suffice
```
it('should render correct columns', async () => {
    await act(async () => {
      render(
        <AccountTable rowData={singleRowData} onClickAsset={() => null} />
      );
    });
    const headers = screen.getAllByRole('columnheader');
    expect(headers).toHaveLength(6);
    expect(
      headers?.map((h) => h.querySelector('[ref="eText"]')?.textContent?.trim())
    ).toEqual(['Asset', 'Deposited', 'Used', '', '', '']);
  });
```

File: libs/accounts/src/lib/accounts-table.spec.tsx [hunk 22]
```diff
@@ -24,47 +25,104 @@ const singleRow: AccountFieldsFragment = {
...
-        ];
-        cells.forEach((cell, i) => {
-          expect(cell).toHaveTextContent(expectedValues[i]);
-        });
+    await act(async () => {
+      render(
+        <AccountTable rowData={singleRowData} onClickAsset={() => null} />
+      );
+    });
+    await waitFor(async () => {
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-table.spec.tsx:66]: chore (blocking): `waitFor` shouldn't be required here.

File: libs/accounts/src/lib/accounts-table.tsx [hunk 23]
```diff
@@ -1,172 +1,204 @@
...
+        >
+          <AgGridColumn
+            headerName={t('Asset')}
+            field="asset.symbol"
+            headerTooltip={t(
+              'Asset is the collateral that is deposited into the Vega protocol.'
+            )}
+            cellRenderer={({
+              value,
+            }: VegaValueFormatterParams<AccountFields, 'asset.symbol'>) => {
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-table.tsx:97]: suggestion (non blocking): use `VegaICellRendererParams`

File: libs/accounts/src/lib/accounts-table.tsx [hunk 24]
```diff
@@ -1,172 +1,204 @@
...
+            cellRendererSelector={progressBarCellRendererSelector}
+            valueFormatter={progressBarValueFormatter}
+          />
+          <AgGridColumn
+            headerName=""
+            field="breakdown"
+            maxWidth={150}
+            cellRenderer={({
+              value,
+            }: VegaICellRendererParams<AccountFields, 'breakdown'>) => {
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-table.tsx:139]: suggestion (non blocking): `VegaICellRendererParams` (and elsewhere in this file.

File: libs/accounts/src/lib/breakdown-table.spec.tsx [hunk 25]
```diff
@@ -0,0 +1,120 @@
...
+  },
+  available: '125600000',
+  used: '125600000',
+  deposited: '125600000',
+};
+const singleRowData = [singleRow];
+
+describe('BreakdownTable', () => {
+  it('should render successfully', async () => {
+    const { baseElement } = render(<BreakdownTable data={[]} />);
```
Comment (mattrussell36) [inline libs/accounts/src/lib/breakdown-table.spec.tsx:36]: chore (blocking): Needs act to remove console.error in test output

File: libs/accounts/src/lib/breakdown-table.spec.tsx [hunk 26]
```diff
@@ -0,0 +1,120 @@
...
+    expect(baseElement).toBeTruthy();
+  });
+
+  it('should render correct columns', async () => {
+    await act(async () => {
+      render(<BreakdownTable data={singleRowData} />);
+    });
+    await waitFor(async () => {
```
Comment (mattrussell36) [inline libs/accounts/src/lib/breakdown-table.spec.tsx:44]: chore (blocking): redundant waitFor (and in other tests)

File: libs/accounts/src/lib/breakdown-table.tsx [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+          valueFormatter={({
+            value,
+          }: VegaValueFormatterParams<AccountFields, 'type'>) =>
+            AccountTypeMapping[value]
+          }
+        />
+        <AgGridColumn
+          headerName={t('Market')}
+          field="market.tradableInstrument.instrument.name"
+          valueFormatter="value || '—'"
```
Comment (mattrussell36) [inline libs/accounts/src/lib/breakdown-table.tsx:57]: question (non blocking): Does AgGrid allows passing a string here? I'd prefer to write it with proper js so we get type safety

File: libs/accounts/src/lib/breakdown-table.tsx [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+        />
+
+        <AgGridColumn
+          headerName={t('Deposited')}
+          field="deposited"
+          valueFormatter={({
+            value,
+            data,
+          }: VegaValueFormatterParams<AccountFields, 'deposited'>) =>
+            data &&
```
Comment (mattrussell36) [inline libs/accounts/src/lib/breakdown-table.tsx:77]: suggestion (non blocking): Prefer non implicit return and an if statement. Also do we not want to return `'-'` if there is no data to be more consistent with other columns and tables?

File: libs/withdraws/src/lib/withdraw-manager.spec.tsx [hunk 29]
```diff
@@ -23,77 +47,84 @@ jest.mock('./use-get-withdraw-delay', () => ({
...
-    amount: '1000',
-    asset: props.assets[0].id,
-    receiverAddress: ethereumAddress,
-    availableTimestamp: null,
+  it('calls submit if valid form submission', async () => {
+    render(generateJsx(props));
+    await act(async () => {
+      await submitValid();
+    });
+    await waitFor(async () => {
```
Comment (mattrussell36) [inline libs/withdraws/src/lib/withdraw-manager.spec.tsx:71]: chore (blocking): Some redundant code here can be tidied up, no need for waitFor, no need to await props.submit

```
render(generateJsx(props));
await act(async () => {
  await submitValid();
});
expect(props.submit).toHaveBeenCalledWith({
  amount: '1000',
  asset: props.assets[0].id,
  receiverAddress: ethereumAddress,
  availableTimestamp: null,
});
```

File: libs/accounts/src/lib/accounts-table.tsx [hunk 30]
```diff
@@ -1,172 +1,204 @@
...
+              return (
+                <ButtonLink
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-table.tsx:141]: suggestion (non blocking): Can we just make all the buttons in the table consistent for now. There was a discussion on how to render buttons in tables but no decision has been made yet, so lets just keep it the same as the others

File: libs/accounts/src/lib/accounts-data-provider.ts [hunk 31]
```diff
@@ -67,3 +71,82 @@ export const accountsDataProvider = makeDataProvider<
...
+    asset: accounts[0].asset,
+    balance: available.toString(),
+    type: AccountType.ACCOUNT_TYPE_GENERAL,
+    available: available.toString(),
+    used: used.toString(),
+    deposited: (available + used).toString(),
+  };
+
+  const breakdown = accounts
+    .filter((a) => AccountType.ACCOUNT_TYPE_GENERAL !== a.type)
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-data-provider.ts:130]: suggestion (non blocking): I think this code would be more readable if you filter'd in the USE_ACCOUNT_TYPES and then just mapped over them. Then I don't think you need the addtional ternary checks within the map callback itself

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/accounts/src/lib/accounts-table.tsx (hunk 30 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx:141]: Pete suggested to make `Collateral breakdown` ButtonLinks

File: libs/accounts/src/lib/accounts-data-provider.ts (hunk 31 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-data-provider.ts:130]: The other day I received a comment that `AccountType.ACCOUNT_TYPE_GENERAL !== a.type` would make it more readable. Plus, I want to display as much as I can in the breakdown, almost every account entry except what is in general.

File: libs/withdraws/src/lib/withdraw-manager.spec.tsx (hunk 29 above)
Comment (MadalinaRaicu) [reply libs/withdraws/src/lib/withdraw-manager.spec.tsx]: updated

File: libs/accounts/src/lib/breakdown-table.tsx (hunk 28 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/breakdown-table.tsx]: updated

File: libs/accounts/src/lib/breakdown-table.tsx (hunk 27 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/breakdown-table.tsx]: updated

File: libs/accounts/src/lib/breakdown-table.spec.tsx (hunk 26 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/breakdown-table.spec.tsx]: updated

File: libs/withdraws/src/lib/use-withdraw-asset.tsx [hunk 32]
```diff
@@ -0,0 +1,58 @@
...
+      let threshold;
+      let delay;
+      try {
+        const result = await Promise.all([getThreshold(asset), getDelay()]);
+        threshold = result[0];
+        delay = result[1];
+      } catch (err) {
+        captureException(err);
+      }
+      update({ asset, balance, min, threshold, delay });
```
Comment (asiaznik) [inline libs/withdraws/src/lib/use-withdraw-asset.tsx:49]: question (non-blocking): because threshold is of type `BigNumber` it will get updated to `undefined` when failed - isn't it breaking anything?

File: libs/accounts/src/lib/breakdown-table.spec.tsx (hunk 25 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/breakdown-table.spec.tsx:36]: updated

File: libs/accounts/src/lib/accounts-table.tsx (hunk 23 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.tsx]: updated

File: libs/accounts/src/lib/accounts-table.spec.tsx (hunk 22 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.spec.tsx]: removed all waitFor

File: libs/accounts/src/lib/accounts-table.spec.tsx (hunk 21 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.spec.tsx]: updated

File: apps/console-lite/src/app/components/console-lite-grid/console-lite-grid.tsx [hunk 33]
```diff
@@ -26,8 +26,10 @@ interface Props<T> extends GridOptions {
   classNamesParam?: string | string[];
 }
 
-const ConsoleLiteGrid = <T extends { id?: string }>(
-  { data, handleRowClicked, getRowId, classNamesParam, ...props }: Props<T>,
+const ConsoleLiteGrid = <T,>(
```
[the remark this answers is not in the document]
Comment (MadalinaRaicu) [reply apps/console-lite/src/app/components/console-lite-grid/console-lite-grid.tsx]: this line is outdated. updated in another commit.

File: libs/accounts/src/lib/accounts-table.spec.tsx (hunk 20 above)
Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-table.spec.tsx]: updated

File: libs/withdraws/src/lib/use-withdraw-asset.tsx (hunk 32 above)
Comment (MadalinaRaicu) [reply libs/withdraws/src/lib/use-withdraw-asset.tsx]: I haven't changed this bit too much, but I agree, this seems a bit off, so I am moving update inside the try.

Comment (MadalinaRaicu) [reply libs/withdraws/src/lib/use-withdraw-asset.tsx]: updated

**Review by mattrussell36 — APPROVED** [review]
